test/trafficGenerator/SpawnScheduler.spec.js

```diff
@@ -116,6 +116,53 @@ ava('.resetTimer() returns early when SpawnPatternModel has no #scheduleId', (t)
     t.true(destroyTimerStub.notCalled);
 });
 
+ava('.resetTimer() does not destroy existing timers when SpawnPatternModel has a zero #scheduleId', (t) => {
+    SpawnScheduler.init(aircraftControllerStub);
+    const destroyTimerStub = sandbox.stub(GameController, 'destroyTimer');
+    const spawnPatternModel = SpawnPatternCollection._items[0];
+    spawnPatternModel.scheduleId = 0;
+
+    SpawnScheduler.resetTimer(spawnPatternModel);
+
+    delete spawnPatternModel.scheduleId;
+
+    SpawnScheduler.resetTimer(spawnPatternModel);
+
+    t.true(destroyTimerStub.notCalled);
+});
+
+ava('.resetTimer() destroys existing timers when SpawnPatternModel has a positive and valid #scheduleId', (t) => {
+    SpawnScheduler.init(aircraftControllerStub);
+    const destroyTimerStub = sandbox.stub(GameController, 'destroyTimer');
+    const spawnPatternModel = SpawnPatternCollection._items[0];
+    spawnPatternModel.scheduleId = 10;
+
+    SpawnScheduler.resetTimer(spawnPatternModel);
+
+    delete spawnPatternModel.scheduleId;
+
+    SpawnScheduler.resetTimer(spawnPatternModel);
+
+    t.true(destroyTimerStub.calledOnce);
+});
+
+ava('.resetTimer() destroys existing timers twice when SpawnPatternModel has a positive and valid #scheduleId and rate is set to 0 and then positive', (t) => {
+    SpawnScheduler.init(aircraftControllerStub);
```

Comment:
you should also check that:
- no timer will be created when rate is 0
- a new timer will be created when rate > 0

Reply:
@felixscheffer. I agree test case must take into account varous rates.

Reply:
Test case improvement commited.

Reply:
Time delta calculation testing was not in scope of this issue. I'll check into it and open another issue.

Reply:
Time delta calculation test covererage is 100% in test/trafficGenerator/SpawnPatternModel.spec.js.
I don't think there's a need for further unit testing.